clients/drcachesim/tracer/raw2trace.h

```diff
@@ -1442,6 +1455,75 @@ template <typename T> class trace_converter_t {
         return "";
     }
 
+#ifdef BUILD_PT_POST_PROCESSOR
+#    define PT_DATA_FILE_NAME_SUFFIX ".pt"
+#    define PT_METADAT_FILE_NAME_SUFFIX ".metadata"
+#    define KERNEL_IMAGE_FILE_NAME "kimage"
+#    define KERNEL_IMAGE_METADATA_FILE_NAME "kimage.metadata"
+    std::string
+    append_syscall_pt_trace(void *tls, thread_id_t thread_id, uint64_t syscall_id)
+    {
+        std::string error = "";
+        pt2ir_config_t config = {};
+        config.raw_file_path = impl()->get_syscall_pt_trace_dir(tls) + "/" +
+            std::to_string(thread_id) + "." + std::to_string(syscall_id) +
+            PT_DATA_FILE_NAME_SUFFIX;
+        config.kernel_image_path =
+            impl()->get_syscall_pt_trace_dir(tls) + "/" + KERNEL_IMAGE_FILE_NAME;
+        config.kernel_image_metadata_path =
+            impl()->get_syscall_pt_trace_dir(tls) + "/" + KERNEL_IMAGE_METADATA_FILE_NAME;
+        config.init_with_metadata(config.raw_file_path + PT_METADAT_FILE_NAME_SUFFIX);
+
+        std::unique_ptr<pt2ir_t> ptconverter(new pt2ir_t());
+        if (!ptconverter->init(config)) {
+            error = "Failed to initialize pt2ir";
+            return error;
+        }
+        instrlist_autoclean_t drir = { GLOBAL_DCONTEXT, nullptr };
+        pt2ir_convert_status_t pt2ir_convert_status = ptconverter->convert(drir);
+        if (pt2ir_convert_status != PT2IR_CONV_SUCCESS) {
+            error = "Failed to convert PT raw trace to DR IR";
+            return error;
+        }
+
+        std::vector<trace_entry_t> entries;
+        ir2trace_convert_status_t ir2trace_convert_status =
+            ir2trace_t::convert(drir, entries);
+        if (ir2trace_convert_status != IR2TRACE_CONV_SUCCESS) {
+            error = "Failed to convert DR IR to trace entries";
+            return error;
+        }
+
+        trace_entry_t *buf_base = impl()->get_write_buffer(tls);
+        byte *buf = reinterpret_cast<byte *>(buf_base);
+        for (auto &entry : entries) {
+            trace_entry_t *buf_entry = reinterpret_cast<trace_entry_t *>(buf);
+            *buf_entry = entry;
+            buf += sizeof(trace_entry_t);
```

Comment:
A trace analyzer may assert that load/store instructions have data address entries immediately afterward.  We need a solution to indicate what to expect.  The offline file type mentioned above helps: but it would be better to delimit the instr-only regions.  A new marker type at the start and end?  A general type that doesn't care about kernel but is about instr-only, to support using PT for windows of user code too.  Or it shouldn't really be instr-only as we plan to synthesize data entries: but on that note, do we consider this instr-only state a temporary non-published state (and so should have comments here making that clear) and we only want to export PT sequences that have synthesized data entries?